pkg/clustermanager/cluster_manager.go

```diff
@@ -708,16 +708,23 @@ func (c *ClusterManager) waitForControlPlaneReplicasReady(ctx context.Context, m
 }
 
 func (c *ClusterManager) waitForMachineDeploymentReplicasReady(ctx context.Context, managementCluster *types.Cluster, clusterSpec *cluster.Spec) error {
+	var err error
 	var machineDeploymentReplicasCount int
+	readyReplicas, totalReplicas := 0, 0
+	policy := func(_ int, _ error) (bool, time.Duration) {
```

Comment:
Why define an inline function with parameters and not use them?

Reply:
I see why.

Can we move this down and define it in the `retrier.WithRetryPolicy()`?